wpiutil/src/main/native/thirdparty/llvm/include/wpi/StringExtras.h

```diff
@@ -538,6 +540,52 @@ constexpr std::pair<std::string_view, std::string_view> rsplit(
           slice(str, idx + separator.size(), std::string_view::npos)};
 }
 
+/**
+ * Splits @p str into substrings around the occurrences of a separator string.
+ *
+ * Each substring is passed to the callback @p func. If @p maxSplit is >= 0, at
+ * most @p maxSplit splits are done and consequently <= @p maxSplit + 1
+ * elements are added to arr.
+ * If @p keepEmpty is false, empty strings are not included. They
+ * still count when considering @p maxSplit
+ * An useful invariant is that
+ * separator.join(all callbacks) == str if maxSplit == -1 and keepEmpty == true
+ *
+ * @param separator The string to split on.
+ * @param maxSplit The maximum number of times the string is split.
+ * @param keepEmpty True if empty substring should be added.
+ * @param func Function to call for each substring.
+ */
+template <typename F>
```

Comment:
Just to double check, does it matter to constrain `F` to accept a `std::string_view`?

Reply:
Yes, I'm planning on adding constraints here.